core/trino-main/src/test/java/io/trino/sql/planner/AbstractPredicatePushdownTest.java

```diff
@@ -66,14 +66,14 @@ public void testNonStraddlingJoinExpression()
         assertPlan(
                 "SELECT * FROM orders JOIN lineitem ON orders.orderkey = lineitem.orderkey AND cast(lineitem.linenumber AS varchar) = '2'",
                 anyTree(
-                        join(INNER, ImmutableList.of(equiJoinClause("LINEITEM_OK", "ORDERS_OK")),
+                        join(INNER, ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
```

Comment:
Join rotated. cc @sopel39

Reply:
> Join rotated.

Because of CBO. Do we get better stats now?

Reply:
`ScalarStatsCalculator` should return same result for both forms (`VARCHAR 'abc'` and `CAST('abc' AS varchar)`). However, we have about 15 cases of `instanceof Literal`, which can be activated by the change.

Reply:
So are stats better now?